addons/refuel/functions/fnc_connectNozzleAction.sqf

```diff
@@ -0,0 +1,109 @@
+/*
+ * Author: GitHawk et.al
+ * Calculates a connection for tilting
+ * With code from ace_attach
+ *
+ * Arguments:
+ * 0: The player <OBJECT>
+ * 1: The target <OBJECT>
+ * 2: The visual position <ARRAY>
+ * 3: The nozzle <OBJECT>
+ *
+ * Return Value:
+ * NIL
+ *
+ * Example:
+ * [player, tank, [0,0,0], nozzle] call ace_refuel_fnc_connectNozzleAction
+ *
+ * Public: Yes
+ */
+#include "script_component.hpp"
+#define FLOWRATE_GND 0.165 // TODO ace_vehicles messes with fuel capacity -  why?
+#define FLOWRATE_AIR 0.67
+#define RATE 1 // FIXME use global var from module
+
+private ["_startingOffset", "_startDistanceFromCenter", "_closeInUnitVector", "_closeInMax", "_closeInMin", "_closeInDistance", "_endPosTestOffset", "_endPosTest", "_doesIntersect", "_startingPosShifted", "_endASL", "_rate", "_maxFuel"];
+
+params ["_unit", "_target", "_startingPosition", "_nozzle"];
+_startingOffset = _target worldToModel _startingPosition;
+
+_startDistanceFromCenter = vectorMagnitude _startingOffset;
+_closeInUnitVector = vectorNormalized (_startingOffset vectorFromTo [0,0,0]);
+
+_closeInMax = _startDistanceFromCenter;
+_closeInMin = 0;
+
+while {(_closeInMax - _closeInMin) > 0.01} do {
+    _closeInDistance = (_closeInMax + _closeInMin) / 2;
+    _endPosTestOffset = _startingOffset vectorAdd (_closeInUnitVector vectorMultiply _closeInDistance);
+    _endPosTestOffset set [2, (_startingOffset select 2)];
+    _endPosTest = _target modelToWorldVisual _endPosTestOffset;
+
+    _doesIntersect = false;
+    {
+        if (_doesIntersect) exitWith {};
+        _startingPosShifted = _startingPosition vectorAdd _x;
+        _startASL = if (surfaceIsWater _startingPosShifted) then {_startingPosShifted} else {ATLtoASL _startingPosShifted};
+        {
+            _endPosShifted = _endPosTest vectorAdd _x;
+            _endASL = if (surfaceIsWater _startingPosShifted) then {_endPosShifted} else {ATLtoASL _endPosShifted};
+
+            //Uncomment to see the lazor show, and see how the scanning works:
+            // drawLine3D [_startingPosShifted, _endPosShifted, [1,0,0,1]];
+            if (_target in lineIntersectsWith [_startASL, _endASL, _unit]) exitWith {_doesIntersect = true};
+        } forEach [[0,0,0.045], [0,0,-0.045], [0,0.045,0], [0,-0.045,0], [0.045,0,0], [-0.045,0,0]];
```

Comment:
Good news is that in 1.50 there is going to be an even better command that will improve a lot of this.
So don't worry about optimizing it now. We should be able to drop in the new way once 1.50 is out.